### PR TITLE
Feat/add proper logging

### PR DESCRIPTION
**Pull request by HaidarJbeily7**
related to this [Issue](https://github.com/objectionary/eo2js/issues/85)

---

## PR-Codex overview
This PR introduces additional logging throughout various command files in the `eo2js` project to enhance visibility into the execution flow, making it easier to debug and understand the processes involved in dataizing, testing, linking, and transpiling.

### Detailed summary
- Added logging for the object being dataized and the main file used in `dataize.js`.
- Logged the directory where tests are running and any excluded patterns in `test.js`.
- Logged the creation of `package.json` and the copying of dependencies in `link.js`.
- Logged the copying of main files and completion of project builds in `link.js`.
- Logged the reading of foreign files and the output directory in `transpile.js`.
- Logged the number of verified tojos found and processed in `transpile.js`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo2js/src/commands/dataize.js [hunk 1]
```diff
@@ -11,6 +11,11 @@ const {execSync} = require('child_process')
 const dataize = function(obj, args, options) {
   options = {...program.opts(), ...options}
   const main = path.resolve(options.target, options.project, '__main__.js')
+  console.log(`Dataizing object: ${obj}`)
+  console.log(`Using main file: ${main}`)
```
Comment (maxonfjvipon) [inline eo2js/src/commands/dataize.js]: @HaidarJbeily7 let's merge these two `console.log`s into single one

File: eo2js/src/commands/link.js [hunk 2]
```diff
@@ -52,22 +52,33 @@ const pckg = function(options) {
 const link = function(options) {
   options = {...program.opts(), ...options}
   const project = path.resolve(options.target, options.project)
+  console.log(`Building npm project in: ${project}`)
+
+  console.log('Creating package.json...')
```
Comment (maxonfjvipon) [inline eo2js/src/commands/link.js]: @HaidarJbeily7 let's move this log to 3 lines below and print something like: `created package.json in ...` where ... is path

File: eo2js/src/commands/link.js [hunk 3]
```diff
@@ -52,22 +52,33 @@ const pckg = function(options) {
...
   fs.writeFileSync(
     path.resolve(project, 'package.json'),
     JSON.stringify(pckg(options))
   )
+
+  console.log('Installing dependencies...')
   execSync('npm install', {cwd: project})
+
+  console.log(`Copying ${main} file...`)
```
Comment (maxonfjvipon) [inline eo2js/src/commands/link.js]: @HaidarJbeily7 let's move this log below after copying is done and print something like: `copied file to directory`

File: eo2js/src/commands/link.js [hunk 4]
```diff
@@ -52,22 +52,33 @@ const pckg = function(options) {
...
   fs.copyFileSync(
     path.resolve(options.resources, `js/${main}`),
     path.resolve(project, main)
   )
+
   if (options.dependency) {
+    console.log(`Copying local eo2js-runtime from: ${options.dependency}`)
```
Comment (maxonfjvipon) [inline eo2js/src/commands/link.js]: @HaidarJbeily7 same as above

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

**Comment by HaidarJbeily7** [comment]
Closes: #85

**Review by maxonfjvipon — APPROVED** [review]
@HaidarJbeily7 LGTM

**Comment by maxonfjvipon** [comment]
@rultor merge

**Comment by maxonfjvipon** [comment]
@HaidarJbeily7 thanks

**Pull request merged**